src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Javac.targets

```diff
@@ -0,0 +1,137 @@
+        DefaultJdkVersion="$(_DefaultJdkVersion)"
+        SkipJavacVersionCheck="$(AndroidSkipJavacVersionCheck)"
+        EnableProguard="$(AndroidEnableProguard)"
+        EnableMultiDex="$(AndroidEnableMultiDex)">
+      <Output TaskParameter="TargetVersion" PropertyName="JavacTargetVersion" />
+      <Output TaskParameter="SourceVersion" PropertyName="JavacSourceVersion" />
+    </AdjustJavacVersionArguments>
+  </Target>
+
+  <Target Name="_DetermineBindingJavaLibrariesToCompile" Condition=" '@(_JavaBindingSource->Count())' != '0' ">
+    <DetermineJavaLibrariesToCompile
+        MonoPlatformJarPaths="$(MonoPlatformJarPath);$(_RuntimeJar)"
+        JavaSourceFiles="@(_JavaBindingSource)"
+        JavaLibraries="@(AndroidJavaLibrary);@(EmbeddedJar)"
+        ExternalJavaLibraries="@(AndroidExternalJavaLibrary)"
+        LibraryProjectJars="@(ExtractedJarImports)"
+        DoNotPackageJavaLibraries="@(_ResolvedDoNotPackageAttributes)"
+        EnableInstantRun="$(_InstantRunEnabled)">
+      <Output TaskParameter="JavaLibrariesToCompile" ItemName="_BindingJavaLibrariesToCompile" />
+      <Output TaskParameter="ReferenceJavaLibraries" ItemName="_ReferenceJavaLibs" />
+    </DetermineJavaLibrariesToCompile>
+  </Target>
+
+  <Target Name="_DetermineJavaLibrariesToCompile">
+
+    <DetermineJavaLibrariesToCompile
+        MonoPlatformJarPaths="$(MonoPlatformJarPath);$(_RuntimeJar)"
+        JavaSourceFiles="@(AndroidJavaSource)"
+        JavaLibraries="@(AndroidJavaLibrary)"
+        ExternalJavaLibraries="@(AndroidExternalJavaLibrary)"
+        LibraryProjectJars="@(ExtractedJarImports)"
+        DoNotPackageJavaLibraries="@(_ResolvedDoNotPackageAttributes)"
+        EnableInstantRun="$(_InstantRunEnabled)">
+      <Output TaskParameter="JavaLibrariesToCompile" ItemName="_JavaLibrariesToCompile" />
+      <Output TaskParameter="ReferenceJavaLibraries" ItemName="_ReferenceJavaLibs" />
+    </DetermineJavaLibrariesToCompile>
+
+    <ItemGroup>
+      <_JavaLibrariesToCompileForApp Include="@(_JavaLibrariesToCompile)" Condition=" '$(_InstantRunEnabled)' != 'True' " />
+      <AndroidExternalJavaLibrary    Include="@(_JavaLibrariesToCompile)" Condition=" '$(_InstantRunEnabled)' == 'True' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CollectJavaSourceForBinding">
+    <ItemGroup>
+      <_JavaBindingSource Include="@(AndroidJavaSource)" Condition=" '%(AndroidJavaSource.Bind)' == 'True' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CollectJavaSource">
+    <ItemGroup>
+      <_JavaSource Include="%(AndroidJavaSource.Identity)" Condition=" '%(AndroidJavaSource.Bind)' != 'True' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CompileBindingJava"
+      Condition=" '@(_JavaBindingSource->Count())' != '0' "
+      DependsOnTargets="$(_CompileBindingJavaDependsOnTargets)"
+      Inputs="@(_AndroidMSBuildAllProjects);$(MonoPlatformJarPath);@(_JavaBindingSource)"
+      Outputs="$(_AndroidIntermediateBindingClassesZip)">
```

Comment:
Is it worth doing a `<Touch/>` on this file? The `_CompileJava` is doing that with a stamp file, maybe it solved a bug and we added that?